src/hp/derp/client.rs

```diff
@@ -715,3 +727,132 @@ pub(crate) fn parse_recv_frame(frame: &[u8]) -> Result<(PublicKey, &[u8])> {
         &frame[PUBLIC_KEY_LENGTH..],
     ))
 }
+
+/// Combines blobs into packets of at most MAX_PACKET_SIZE.
+///
+/// Each item in a packet has a big-endian 2-byte length prefix.
+pub struct PacketizeIter<I: Iterator> {
+    iter: std::iter::Peekable<I>,
+    buffer: BytesMut,
+}
+
+impl<I: Iterator> PacketizeIter<I> {
+    /// Create a new new PacketizeIter from something that can be turned into an
+    /// iterator of slices, like a Vec<Bytes>.
+    pub fn new(iter: impl IntoIterator<IntoIter = I>) -> Self {
+        Self {
+            iter: iter.into_iter().peekable(),
+            buffer: BytesMut::with_capacity(MAX_PACKET_SIZE),
+        }
+    }
+}
+
+impl<I: Iterator> Iterator for PacketizeIter<I>
+where
+    I::Item: AsRef<[u8]>,
+{
+    type Item = Bytes;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        use bytes::BufMut;
+        while let Some(next_bytes) = self.iter.peek() {
+            let next_bytes = next_bytes.as_ref();
+            let next_length: u16 = next_bytes.len().try_into().expect("items < 64k size");
+            if self.buffer.len() + next_length as usize + 2 > MAX_PACKET_SIZE {
+                break;
+            }
+            let length_prefix = next_length.to_be_bytes();
```

Comment:
don't we use `LE` in other places of the protocol?

Reply:
if you want LE you get LE.